### PR TITLE
Hooks run

### PR DESCRIPTION
**Pull request by nopcoder**
Part of #1432

### CODE REVIEW

File: actions/manager.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-multierror"
+	"github.com/treeverse/lakefs/db"
+)
+
+type Manager struct {
+	DB db.Database
+}
+
+type RunnableHook struct {
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply actions/manager.go:17]: need a better name - it holds runnable related context variables:
	RunID  string
	Action *Action
	HookID string

[the remark this answers is not in the document]
Comment (itaiad200) [reply actions/manager.go:17]: `Task` ?

File: cmd/lakefs-loadtest/cmd/entry.go [hunk 2]
```diff
@@ -57,9 +58,10 @@ var entryCmd = &cobra.Command{
 
 		conf := config.NewConfig()
 		c, err := catalog.NewCataloger(catalog.Config{
-			Config: conf,
-			DB:     database,
-			LockDB: lockDB,
+			Config:  conf,
+			DB:      database,
+			LockDB:  lockDB,
+			Actions: actions.New(database),
```
Comment (itaiad200) [inline cmd/lakefs-loadtest/cmd/entry.go:64]: Actions isn't a must for `EntryCataloger`. You can handle the nil case in graveler and avoid passing it from places where it's irrelevant.

File: gateway/playback_test.go [hunk 3]
```diff
@@ -115,8 +115,9 @@ func getBasicHandler(t *testing.T, authService *simulator.PlayBackMockConf) (htt
 
 	conn, _ := testutil.GetDB(t, databaseURI)
 	cataloger, err := catalog.NewCataloger(catalog.Config{
-		Config: config.NewConfig(),
-		DB:     conn,
+		Config:  config.NewConfig(),
+		DB:      conn,
+		Actions: actions.New(conn),
```
Comment (itaiad200) [inline gateway/playback_test.go:120]: Unnecessary.

File: graveler/graveler.go [hunk 4]
```diff
@@ -320,17 +324,8 @@ type VersionController interface {
...
-
-	// SetPreCommitHook set pre-commit hook function
-	SetPreCommitHook(fn PreCommitFunc)
-
-	// PreMergeHook get pre-merge hook function
-	PreMergeHook() PreMergeFunc
-
-	// SetPreMergeHook set pre-merge hook function
-	SetPreMergeHook(fn PreMergeFunc)
+	// SetHooksHandler setup handler for all graveler hooks
```
Comment (itaiad200) [inline graveler/graveler.go:327]:
```suggestion
	// SetHooksHandler set handler for all graveler hooks
```

File: graveler/graveler_test.go [hunk 5]
```diff
@@ -243,8 +295,8 @@ func TestGraveler_CreateBranch(t *testing.T) {
 	gravel := graveler.NewGraveler(branchLocker, nil,
 		nil,
 		&testutil.RefsFake{
-			Err:      graveler.ErrNotFound,
-			CommitID: "8888888798e3aeface8e62d1c7072a965314b4",
+			Err: graveler.ErrNotFound,
+			CommitID: "8888888798e3	aeface8e62d1c7072a965314b4",
```
Comment (itaiad200) [inline graveler/graveler_test.go:299]: what happened here?

File: actions/actions.go [hunk 6]
```diff
@@ -10,66 +12,69 @@ import (
 	"github.com/treeverse/lakefs/db"
 )
 
-type Manager struct {
+type Actions struct {
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply actions/actions.go:15]: Service?

File: actions/action.go [hunk 7]
```diff
@@ -114,22 +115,34 @@ func ParseAction(data []byte) (*Action, error) {
 	return &act, nil
 }
 
-func LoadActions(source Source) ([]*Action, error) {
-	names, err := source.List()
+func LoadActions(ctx context.Context, source Source) ([]*Action, error) {
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply actions/action.go:118]: use test code and it is under `actions_test` package

File: graveler/graveler_test.go (hunk 5 above)
Comment (nopcoder) [reply graveler/graveler_test.go:299]: mistake

File: actions/action_test.go [hunk 8]
```diff
@@ -281,87 +281,3 @@ func TestMatchedActions(t *testing.T) {
 		})
 	}
 }
-
-func TestMatchActions(t *testing.T) {
```
Comment (itaiad200) [inline actions/action_test.go:285]: Why delete?

File: actions/service.go [hunk 9]
```diff
@@ -23,13 +23,13 @@ type Task struct {
...
-func New(db db.Database) *Actions {
-	return &Actions{
+func New(db db.Database) *Service {
+	return &Service{
 		DB: db,
 	}
 }
 
-func (m *Actions) Run(ctx context.Context, event Event) error {
+func (m *Service) Run(ctx context.Context, event Event) error {
```
Comment (itaiad200) [inline actions/service.go:32]: nit
```suggestion
func (s *Service) Run(ctx context.Context, event Event) error {
```

**Review by itaiad200 — APPROVED** [review]
Nice work!!

File: actions/action_test.go (hunk 8 above)
Comment (nopcoder) [reply actions/action_test.go:285]: because I turned it into private as you asked... return it as public

**Pull request merged**